### PR TITLE
feat: added `tryParseURL`

### PR DESCRIPTION
**Pull request by legendhimslef**
fixes: #481 
Todo:
- [x] Write unit test

### CODE REVIEW

File: packages/utilities/package.json [hunk 1]
```diff
@@ -262,9 +262,19 @@
 			"require": "./dist/lib/toTitleCase.js"
 		},
 		"./tryParse": {
```
Comment (vladfrangu) [inline packages/utilities/package.json:264]: Should probably remove this

File: packages/utilities/src/lib/tryParseJSON.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+/**
+ * Try parse a stringified JSON string.
+ * @param value The value to parse
+ */
+export function tryParseJSON(value: string): object | string {
```
Comment (vladfrangu) [inline packages/utilities/src/lib/tryParseJSON.ts:5]: Should return whatever json.parse says it returns in typings

File: packages/utilities/src/lib/tryParseJSON.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+	try {
+		return JSON.parse(value);
+	} catch (err) {
+		return value;
+	}
+}
+
+/**
```
Comment (vladfrangu) [inline packages/utilities/src/lib/tryParseJSON.ts:13]: @favna do we need to do this or can we just axe it ?

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: packages/utilities/package.json (hunk 1 above)
Comment (imranbarbhuiya) [reply packages/utilities/package.json:264]: it'll be a breaking change ig?

File: packages/utilities/src/lib/tryParseJSON.ts (hunk 2 above)
Comment (imranbarbhuiya) [reply packages/utilities/src/lib/tryParseJSON.ts:5]: `JSON.parse` return type is `any`

File: packages/utilities/src/lib/tryParseJSON.ts (hunk 3 above)
Comment (favna) [reply packages/utilities/src/lib/tryParseJSON.ts:13]: Yes, removing functions is a breaking change

File: packages/utilities/package.json (hunk 1 above)
Comment (favna) [reply packages/utilities/package.json:264]: Indeed, removing functions is a breaking change

**Review by favna — APPROVED** [review]
I'll be honest I had completely missed that we already had `parseURL` because I was looking for `try...` when I created the issue LMAO!! 

Ohwell, ultimately we also got general improvements out of this PR

**Comment by imranbarbhuiya** [comment]
@favna I was just checking if there is any way to add deprecated jsdoc in re-export and found it 

https://github.com/microsoft/TypeScript/issues/42684

**Comment by favna** [comment]
> @favna I was just checking if there is any way to add deprecated jsdoc in re-export and found it
> 
> [microsoft/TypeScript#42684](https://github.com/microsoft/TypeScript/issues/42684)

Oh can you make a PR for that then? Same for `filterNullishOrZero` and `filterNullishOrEmpty` (mind the OR, so not the AND variants)

**Pull request merged**